photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/GameTrainingDriver.scala

```diff
@@ -660,7 +660,18 @@ object GameTrainingDriver extends GameDriver {
         val (_, baseConfig, evaluationResults) = models.head
 
         val iteration = getOrDefault(hyperParameterTuningIter)
-        val dimension = baseConfig.toSeq.length
+
+        val dimension = baseConfig.toSeq.map {
+          case (_, config: GLMOptimizationConfiguration) =>
+            config.regularizationContext.regularizationType match {
+              case RegularizationType.ELASTIC_NET => 2
+              case RegularizationType.L2 => 1
+              case RegularizationType.L1 => 1
+              case RegularizationType.NONE => 0
+            }
+          case _ => throw new IllegalArgumentException(s"Unknown optimization config!")
```

Comment:
Can you change the error message to be more explicit? Like "Unknown regularization type."